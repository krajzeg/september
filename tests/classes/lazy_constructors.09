Delayed := Class.new("Delayed")
Delayed:::"<constructor>" = |?operation| {
	this::operation = operation
}
Delayed:::run = {
	this.operation.resolve()
}


x := 40
increment := Delayed(x = x + 1)
print("The operation did not run, so x should be 40:", x)

increment.run()
increment.run()
print("The operation has run twice, so x should be 42:", x)
