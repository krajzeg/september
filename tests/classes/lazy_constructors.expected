The operation did not run, so x should be 40: 40
The operation has run twice, so x should be 42: 42
